### PR TITLE
fix(ShadowDomStrategy): always inline import rules

### LINKED ISSUE #1694: Wait for included css to be fully loaded before instantiating a component / finishing the compilation
Use case:

A component needs to know how high it is. Right now it would schedule a promise / eventually use the new `size` event (see #1693). However, if we don't wait for nested CSS to be fully loaded before we instantiate a component the component has no way of knowing the right size.

The problem probably only appears for `NativeShadowDomStrategy`, as in the emulated mode we load nested stylesheets manually.

### PR DESCRIPTION
**Pull request by vicb**
fixes #1694

### CODE REVIEW

**Comment by vicb** [comment]
@naomiblack @tbosch 

This PR contains a fix for #1694 on the current code base because it is high priority for GT.

Some tests are failing because relative URL are not yet supported in templates (that's the pending #2593 which won't be ready before next week) - and it might also be related to the workaround @jelbourn put in place to be able to test the material examples.

I'm wondering if GT does not use a custom `UrlResolver` which supports relative URL resolution ? If this is the case, we might want to merge this by disabling the failing tests for now (and re-enable them once 2593 is ready).

**Comment by yjbanov** [comment]
Why do we need relative URL support for inlining? The two should be orthogonal concerns.

File: modules/angular2/src/render/dom/shadow_dom/emulated_scoped_shadow_dom_strategy.ts [hunk 1]
```diff
@@ -27,8 +27,11 @@ import {
  * - see `ShadowCss` for more information and limitations.
  */
 export class EmulatedScopedShadowDomStrategy extends EmulatedUnscopedShadowDomStrategy {
-  constructor(public styleInliner: StyleInliner, styleUrlResolver: StyleUrlResolver, styleHost) {
-    super(styleUrlResolver, styleHost);
+  styleInliner: StyleInliner;
```
Comment (yjbanov) [inline modules/angular2/src/render/dom/shadow_dom/emulated_scoped_shadow_dom_strategy.ts]: Do you need this field? The super-class already has it.

**Comment by yjbanov** [comment]
superseded by https://github.com/angular/angular/pull/2654

**Comment by vicb** [comment]
@yjbanov thanks for looking at this & fixing

**Pull request closed without merging**